featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -789,23 +809,27 @@ def _feature_in_relationship_path(self, relationship_path, feature):
             return False
 
         for _, relationship in relationship_path:
-            if relationship.child_dataframe.id == feature.entity.id and \
-               relationship.child_column.id == feature.variable.id:
+            if relationship.child_name == feature.dataframe_name and \
+               relationship.child_column.name == feature.column_name:
                 return True
 
-            if relationship.parent_dataframe.id == feature.entity.id and \
-               relationship.parent_column.id == feature.variable.id:
+            if relationship.parent_name == feature.dataframe_name and \
+               relationship.parent_column.name == feature.column_name:
                 return True
 
         return False
 
-    def _get_matching_inputs(self, all_features, entity, max_depth, input_types,
+    def _get_matching_inputs(self, all_features, dataframe, max_depth, input_types,
                              primitive, primitive_options, require_direct_input=False,
                              feature_filter=None):
+
         features = self._features_by_type(all_features=all_features,
-                                          entity=entity,
+                                          dataframe=dataframe,
                                           max_depth=max_depth,
-                                          variable_type=set(input_types))
+                                          # variable_type=set(input_types))
+                                          # TODO: Investigate whether this needs to be a set?
+                                          column_schemas=list(input_types))
```

Comment:
ColumnSchema objects don't currently define a hash function, so they can't be placed in a set. Things seem to work fine without using a set here, but if we need or want a set we will need to add `__hash__` to ColumnSchema.